### PR TITLE
Implement Time Series Forecast (TSF) Indicator

### LINKED ISSUE #6988: Implements Time Series Forecast (TSF)
#### Expected Behavior
Lean support the Time Series Forecast (TSF) Indicator.

#### Actual Behavior
Not supported. See #83.

Source: https://tulipindicators.org/tsf

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by krutik2966**
#### Description

Implements Time Series Forecast (TSF) Indicator.

#### Related Issue

Closes #6988 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Tests/QuantConnect.Tests.csproj [hunk 1]
```diff
@@ -290,6 +290,9 @@
     <None Update="TestData\spy_with_hilbert.csv">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Update="TestData\spy_tsf.csv">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (jaredbroad) [inline Tests/QuantConnect.Tests.csproj:294]: Welcome to LEAN @krutik2966! Please attach this file with the commit so we can run the tests.

Where did you source the data from? It needs to be a reliable test dataset.

Comment (krutik2966) [reply Tests/QuantConnect.Tests.csproj:294]: Thank you so much @jaredbroad!

I've only used the example data provided at [https://tulipindicators.org/tsf](https://tulipindicators.org/tsf). So far, I've only applied the indicator to this specific dataset and haven't tested it on any other data sources. At first, I was extracting the data from this file, but in the final version of the code, I decided to directly put the data into the `Tests/Indicators/TimeSeriesForecastTests.cs` file.

Comment (jaredbroad) [reply Tests/QuantConnect.Tests.csproj:294]: Thank you. Please revert and follow the pattern of other indicator tests by providing the spy_tsf test file

**Comment by Martin-Molinero** [comment]
Closing for now, as per comments above.
Suggest taking a look at the PRs associated with adding new indicators, to understand what's expected, see https://github.com/QuantConnect/Lean/issues?q=is%3Aissue+indicator+is%3Aclosed+label%3Aindicator.
Thank you!

**Pull request closed without merging**